lib/sass/tree/visitors/perform.rb

```diff
@@ -346,29 +346,47 @@ def visit_mixindef(node)
 
   # Runs a mixin.
   def visit_mixin(node)
+    mixin_name = nil
```

Comment:
If you initialize this to `node.name` here, I think you can avoid `mixin_name || node.name` below.